examples/text/plot_document_classification_20newsgroups.py

```diff
@@ -145,7 +145,7 @@ def size_mb(docs):
     len(data_train.data), data_train_size_mb))
 print("%d documents - %0.3fMB (test set)" % (
     len(data_test.data), data_test_size_mb))
-print("%d categories" % len(categories))
+print("%d categories" % len(target_names))
```

Comment:
shouldn't this be counting only unique values in target_names then? Would `len(set(target_names))` work?

Could you please make sure it actually prints the correct value in all cases?

Reply:
```python
data_train = fetch_20newsgroups(subset='train', random_state=42)

len(data_train.target_names)    # this returns 20

len(set(data_train.target_names))    # this returns 20
```
I think `data_train.target_names` is a list of unique target names. 

The documentation of `fetch_20newsgroups` is not that clear to me with regards to what the function returns:

`
bunch : Bunch object
bunch.data: list, length [n_samples] bunch.target: array, shape [n_samples] bunch.filenames: list, length [n_classes] bunch.DESCR: a description of the dataset.
`

Reply:
I think it would be better to rewrite the docstring of `fetch_20newsgroups` instead so that we don't have to write a comment every time `target_names` is used. This could be done in it's own pull-request.

Reply:
yeah, I guess I'm convinced.

Reply:
Thanks.